packages/components/src/border-box-control/border-box-control/component.tsx

```diff
@@ -134,13 +137,13 @@ const UnconnectedBorderBoxControl = (
 						__experimentalIsRenderedInSidebar={
 							__experimentalIsRenderedInSidebar
 						}
-						size={ size }
+						size={ inputHeight }
 					/>
 				) }
 				<BorderBoxControlLinkedButton
 					onClick={ toggleLinked }
 					isLinked={ isLinked }
-					size={ size }
+					size={ inputHeight }
```

Comment:
I think this would be a good follow-up PR so this one doesn't get too big. And because when applying the diff, I ran into an alignment issue with the link button overlapping: 

<img width="983" alt="Screenshot 2023-11-20 at 12 45 32 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/182f9418-8fc3-400f-b8e0-01b7d141fd6f">

I think we can remove the absolute positioning to fix this, but I can explore that separately.